eos-tech-support/eos-hack-reset-first-contact

```diff
@@ -0,0 +1,42 @@
+#!/bin/bash
+
+flag_file=/var/eos-hack-reset-first-contact
+
+# Check that script was run with superuser privileges.
+if [[ $EUID != 0 ]]; then
+    echo "$0 requires superuser privileges."
+    exit 1
+fi
+
+# Check if flag file exists
+if [ ! -f $flag_file ]; then
+    echo "$0: flag file '$flag_file' does not exist"
+    exit 1
+fi
+
+rm $flag_file
+
+IFS=$'\n'
+for user_row in $(cat /etc/passwd); do
+    user=$(echo $user_row | cut -d : -f 1)
+    user_id=$(echo $user_row | cut -d : -f 3)
+
+    if [[ $user_id -ge 1000 && $user_id -le 1999 ]]; then
+        delete_tries=1
+        while true; do
+            echo "Deleting user $user ($user_id) (try: $delete_tries)"
+            userdel $user
```

Comment:
Is it intentional that you are removing the user but not the actual home directory? This could have very weird/surprising effects - if you recreate user later with the same name, but they have a different UID, they will not own their home directory and then their user account will likely be unusable.

Reply:
Our FAQ for deleting all users uses `userdel -r` https://support.endlessm.com/hc/en-us/articles/360006827311-How-do-I-delete-all-users-from-my-computer-

Reply:
I see, well, I wanted to be not that destructive, most of the time the new user, if you are going to use the same name, is gonna match the ID if that was your only user in the system. But you are right, let's clean it properly.